task/fbc-related-image-check/0.1/fbc-related-image-check.yaml

```diff
@@ -25,25 +25,32 @@ spec:
           cpu: 10m
       script: |
         #!/usr/bin/env bash
+        set -o pipefail
         source /utils.sh
         FAILEDIMAGES=""
 
-        relImgs=$(cat "$(workspaces.workspace.path)/hacbs/fbc-validation/confdir/catalog.yaml" | yq -r '.relatedImages[].image' | sed 's/---//')
-        echo -e "These are related images:\n$relImgs"
-        # cycle through those related images and show outputs
-        for i in ${relImgs// /}
-        do
-          if ! skopeo inspect --no-tags "docker://${i}"; then
-            echo "Skopeo inspect failed on related image: $i"
-            FAILEDIMAGES+="$i, "
-          fi
-        done
-        if [ -z "$FAILEDIMAGES" ]; then
-          HACBS_TEST_OUTPUT="$(make_result_json -r SUCCESS -s 1)"
-          echo "${HACBS_TEST_OUTPUT}" | tee "$(results.HACBS_TEST_OUTPUT.path)"
-        else
-          echo "These images failed inspection: $FAILEDIMAGES"
+        relImgs=$(cat "$(workspaces.workspace.path)/hacbs/fbc-validation/confdir/catalog.yml" | yq -r '.relatedImages[].image' | sed 's/---//')
+        if [ $? -ne 0 ]; then
+          echo "Could not get related images, make sure catalog.yaml exists in FBC fragment image and has valid yaml format."
           HACBS_TEST_OUTPUT="$(make_result_json -r FAILURE -f 1)"
           echo "${HACBS_TEST_OUTPUT}" | tee "$(results.HACBS_TEST_OUTPUT.path)"
-          exit 0
```

Comment:
IMO we can just keep exit 0 here and don't indent everything

Reply:
Changed